components/ModalDeleteButton.js

```diff
@@ -0,0 +1,98 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import ReactDOM from "react-dom";
+import styled from "styled-components";
+
+export default function ModalDeleteButton({
+  showModalDelete,
+  setShowModalDelete,
+  currentId,
+  setEntries,
+  setPopUpSettings,
+  Entry,
+}) {
+  const [isBrowser, setIsBrowser] = useState(false);
+  const router = useRouter();
+
+  useEffect(() => {
+    setIsBrowser(true);
+  }, []);
```

Comment:
instead of using this custom state for checking if you are in the browser, please use the dynamic import from next (where you are importing the Modal). Checkout the documentation for that feature. A dynamically importet component is not rendered on the server.